Cut Prometheus 0.16.0rc1.

juliusv opened the pull request.

juliusv commented:
@fabxc @beorn7 @brian-brazil

brian-brazil commented:
:+1:

beorn7 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,139 @@
+
+All changes:
+
+* [CHANGE] Vendoring is now done via govendor instead of godep.
```
Perhaps mention that we require Go 1.5.x now and the GO15VENDOREXPERIMENT=1 env variable.

juliusv replied:
Done.

fabxc replied:
Using `make` the latter is not a user requirement really.

fabxc commented on CHANGELOG.md:
```diff
@@ -1,3 +1,139 @@
+* [CLEANUP] Switch from `github.com/client_golang/extraction` to
+  `github.com/common/expfmt` and related type cleanups.
+* [CLEANUP] Exit Prometheus when the web server encounters a startup error.
+* [CLEANUP] Simplify Travis CI configuration.
```
Stuff like that has nothing to do with the behavior of the Prometheus application. Personally, I wouldn't list such things here (similarly the .pkgignore thing) – the list is nosiy enough already.

juliusv replied:
Don't feel strongly about this, but can remove.

fabxc commented on CHANGELOG.md:
```diff
@@ -1,3 +1,139 @@
+  gracefully.
+* [ENHANCEMENT] Add IDs to H2 tags on status page to allow anchored linking.
+* [BUGFIX] Fix series file deletion behavior when purging archived series.
+* [BUGFIX] Fix scraper HTTP connection leak when encountering HTTP errors.
```
That was never in 0.15.0 – we shouldn't list fixing of errors we introduced in the development phase.

juliusv replied:
Removing.

fabxc commented on CHANGELOG.md:
```diff
@@ -1,3 +1,139 @@
+* [FEATURE] Implement unary expressions for vector types (`-foo`, `+foo`).
+* [FEATURE] Add console templates for the SNMP exporter.
+* [FEATURE] Make it possible to relabel target scrape query parameters.
+* [FEATURE] Add support for `A` records in DNS service discovery.
```
`A` and `AAAA`

juliusv replied:
Fixing.

fabxc commented on CHANGELOG.md:
```diff
@@ -1,3 +1,139 @@
+  mechanisms in one scrape configuration.
+* [ENHANCEMENT] Handle parser and query evaluation runtime panics more
+  gracefully.
+* [ENHANCEMENT] Add IDs to H2 tags on status page to allow anchored linking.
```
Noise.

juliusv replied:
I'll clarify that this is actually a user-requested feature (to be able to link to them).

juliusv replied:
Oh, actually it is already clarified in the text :)

fabxc commented:
:+1:

juliusv commented:
Addressed all comments again (also removed the vendoring env var explanation again, as we also didn't explain how to build with vendored deps besides `make build` before).

beorn7 commented:
Fair enough, without GO15VENDOREXPERIMENT=1 you can build against your checked out dependencies. Still I think its worth mentioning that we have `vendor` directories now, so the behavior of `go build` changes. That's something that should be mentioned as it is now part of the go tool.

beorn7 commented:
Thanks for listening to me... :-(

juliusv commented:
Ok, let's do that with the next RC or full release.
